app/src/main/java/com/github/orkest/Model/FireStoreDatabaseAPI.kt

```diff
@@ -131,29 +142,97 @@ class FireStoreDatabaseAPI {
 
+        return future
+    }
 
+    /**
+     * @param post the post we want to delete in the database
+     * @return a completable future that completes to true only if post is correctly deleted
+     */
+    fun deletePostInDataBase(post: Post): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val postDocument = getPostCollectionRef(post.username).document(post.date.toString())
 
+        postDocument.delete().addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(it) }
 
+        return future
+    }
 
+    /**
+     * @param username the username of the user we want to get the posts from
+     * @return a completable future of a list of post that match the username
+     */
+    fun getUserPostsFromDataBase(username: String): CompletableFuture<List<Post>>{
+        val future = CompletableFuture<List<Post>>()
+        val usersPosts = getPostCollectionRef(username)
+
+        usersPosts.get().addOnSuccessListener{
+            // Get all posts documents as a list of posts objects
+            val list: MutableList<Post> =  it.toObjects(Post::class.java)
+
+            future.complete(list)
+        } //Propagates the exception in case of exception
+        .addOnFailureListener{
+            future.completeExceptionally(it)
+        }
 
+        return future
+    }
 
+    /**
+     * @param lastConnection the last time the user was connected
+     * @return a completable future of a list of post that were recently published
+     */
+    fun getRecentPostsFromDataBase(lastConnection: LocalDateTime): CompletableFuture<List<Post>> {
+        //TODO: Add the condition on posts belonging to the user's friends
+        val future = CompletableFuture<List<Post>>()
+        val posts = db.collectionGroup("posts")
+
+        val currentTime = LocalDateTime.now()
+
+        //TODO: Make this logic better and refactor it + discuss it with team (corner cases like end of month/end of year not covered)
+        //TODO: See if there is a workaround the interdiction to combine plage queries on different fields
+
+        //Get all posts that were published after the last connection of the user, and that are from the current month
+        posts.whereEqualTo(FieldPath.of("date","year"), currentTime.year)
+            .whereEqualTo(FieldPath.of("date","month"), currentTime.monthValue)
+            .whereGreaterThan(FieldPath.of("date","dayOfMonth"),
+                if (currentTime.year > lastConnection.year &&
+                    currentTime.monthValue > lastConnection.monthValue) 1 else
+                        lastConnection.dayOfMonth).get()
+            .addOnSuccessListener {
+                // Get all posts documents as a list of posts objects
+                val list: MutableList<Post> = it.toObjects(Post::class.java)
+                future.complete(list)
+            } //Propagates the exception in case of exception
```

Comment:
If I understand correctly this code fetches all posts created during the current month and since last connection.
This concerns future sprints, but I think we'll need to implement a code that fetches only last posts from people you follow, and if you don't follow anyone or the people you do follow are inactive then we should fetch any post to just give life to the main feed. 
We could discuss this during the next sprint meeting.

Reply:
Yes, exactly. I agree with you it would be a great feature ! We should add it to the product backlog to work on it in the next sprints